test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java

```diff
@@ -195,9 +213,13 @@ private List<String> readPortsFile(Path file) {
             }
         }
 
-        private void initializeWorkingDirectory() {
+        private void initializeWorkingDirectory(boolean preserverWorkingDirectory) {
             try {
-                IOUtils.deleteWithRetry(workingDir);
+                if (preserverWorkingDirectory == false) {
```

Comment:
nit: use positive logic? `if (preserve) { delete distribution dir } else { delete working dir}`

Reply:
I've inverted this conditional.